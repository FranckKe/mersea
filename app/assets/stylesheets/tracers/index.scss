.off-canvas-wrapper, .off-canvas-wrapper-inner {
height: 100%;
}

.report-link {
    color: red;
    padding-top: 1rem;
}

.traceur-title {
    text-align: center;
    padding-top: 1rem;
}

.tracers-flex-container {
    padding-bottom: 4rem;
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    flex-wrap: wrap;
    background-color: inherit;
    height: 100%;

    .card {
        flex-basis: 150px;
        flex-shrink: 1;
        flex-grow: 1;
        margin: .5rem;
        overflow: hidden;
        box-shadow: 0 2px 5px 0 rgba(0, 0, 0, 0.16), 0 2px 10px 0 rgba(0, 0, 0, 0.12);

        .title {
            line-height: 3rem;
            font-size: 1.5rem;
            font-weight: 300;
        }
        .content {
            padding: .7rem;
            font-weight: 300;
            border-radius: 0 0 2px 2px;
        }
        p {
            margin: 0;
        }
        .action {
            border-top: 1px solid rgba(160, 160, 160, 0.2);
            padding: .7rem;
        }
        a {
            display: block;
            margin-right: 1.3rem;
            transition: color 0.3s ease;
            text-transform: uppercase;
            text-decoration: none;
            font-size: .8rem;
            word-wrap: break-word;
        }
        .image {
            position: relative;
            width: 100%;
            .title {
                position: absolute;
                bottom: 0;
                left: 0;
                padding: 1.3rem;
                color: #fff;
            }
            img {
                border-radius: 2px 2px 0 0;
            }
        }
    }
}