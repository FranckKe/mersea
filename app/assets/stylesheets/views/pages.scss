main {
  height: 100%;
}

.map {
  width: 100%;
  height: 100%;

  .leaflet-bottom.leaflet-right .leaflet-control-layers {
    max-height: calc(100vh - 55px - 74px - 10px - 32px - 5px); // Menu - Zoom Buttons - Bottom Margin - Footer - 5px
    overflow-y: scroll;
  }
}

.leaflet-marker-icon {
  max-width: 64px;
  max-height: 64px;
}

.article {
  width: 100%;
  max-width: 750px;
  margin: auto;
  padding-top: 25px;
  padding-right: 25px;
  padding-left: 25px;

  h1 {
    text-align: center;
  }
}
