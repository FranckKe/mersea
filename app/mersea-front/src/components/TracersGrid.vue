<template>
  <div class="columns is-multiline is-mobile">
    <div class="column is-half-mobile is-one-third-tablet is-one-quarter-desktop" :key="idx" v-for="(tracer, idx) in tracersData()">
      <article class="card">
        <div class="card-image">
          <router-link :to="`/tracers/${tracer.id}`">
          <div class="img-wrapper" :style="`background-image: url('${apiUrl}${tracer.photo}')`">
          </div>
          </router-link>
        </div>
        <div class="card-content">
          <router-link :to="`/tracers/${tracer.id}`">{{ tracer.name }}</router-link>
        </div>
      </article>
    </div>
  </div>
</template>

<script>
export default {
  name: "tracers-grid",
  props: ['tracers'],
  data() {
    return {
      apiUrl: this.$apiUrl,
      tracersData: this.tracers
    }
  }
}
</script>
