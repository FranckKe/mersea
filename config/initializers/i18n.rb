Rails.application.config.i18n.available_locales = %i[fr en es]
Rails.application.config.i18n.default_locale = 'fr'
