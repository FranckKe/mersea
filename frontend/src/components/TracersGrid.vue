<template>
  <div class="columns is-multiline is-mobile">
    <div class="column is-half-mobile is-one-third-tablet is-one-quarter-desktop" :key="idx" v-for="(tracer, idx) in tracers">
      <article class="card">
        <div class="card-image">
          <router-link :to="`/tracers/${tracer.id}`">
          <div class="img-wrapper" :style="`background-image: url('${apiUrl}${tracer.photo}')`">
          </div>
          </router-link>
        </div>
        <div class="card-content">
          <div class="content">
            <p class="title is-4"><router-link :to="`/tracers/${tracer.id}`">{{ tracer.name }}</router-link></p>
            <p>{{ tracer.description }}</p>
            <p><strong>Origin: </strong>{{ tracer.origin }}</p>
            <p><strong>Type: </strong>{{ tracer.kind }}</p>
            <p><strong>Created at: </strong><time datetime="2016-1-1">{{ tracer.created_at | formatDate }}</time></p>
          </div>
        </div>
      </article>
    </div>
  </div>
</template>

<script>
export default {
  name: 'tracers-grid',
  props: ['tracers'],
  data() {
    return {
      apiUrl: this.$apiUrl
    }
  }
}
</script>

<style>
</style>
