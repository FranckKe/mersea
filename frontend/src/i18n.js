import Vue from 'vue'
import VueI18n from 'vue-i18n'

Vue.use(VueI18n)

function loadLocaleMessages() {
  const locales = require.context(
    './locales',
    true,
    /[A-Za-z0-9-_,\s]+\.json$/i
  )
  const messages = {}
  locales.keys().forEach(key => {
    const matched = key.match(/([a-z0-9]+)\./i)
    if (matched && matched.length > 1) {
      const locale = matched[1]
      messages[locale] = locales(key)
    }
  })
  return messages
}

// Detect browser language and strip the "region" language
// en-GB => en
let browserI18n = navigator.language
if (navigator.language.indexOf('-') >= 0)
  browserI18n = navigator.language.substring(0, navigator.language.indexOf('-'))

// Default to english if browser language is not supported
let initializedLocal = ['en', 'es', 'fr'].includes(browserI18n)
  ? browserI18n
  : 'en'

export default new VueI18n({
  locale: initializedLocal,
  fallbackLocale: 'en',
  messages: loadLocaleMessages()
})
