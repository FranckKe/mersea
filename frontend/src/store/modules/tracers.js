import axios from 'axios'

const api = axios.create({
  baseURL: process.env.VUE_APP_API_URL,
  headers: { Accept: 'application/json', 'Content-Type': 'application/json' }
})

const state = {
  tracers: [],
  filteredTracers: [],
  loading: true,
  success: false,
  errors: [],
  displayFormat: 'grid',
  perPage: '10'
}

const getters = {
  getFilteredTracers: state => {
    return state.filteredTracers
  },
  getLoading: state => {
    return state.loading
  },
  getError: state => {
    return state.errors
  },
  getSuccess: state => {
    return state.success
  },
  getTracers: state => {
    return state.tracers
  },
  getPerPage: state => {
    return state.perPage
  },
  getTracerById: state => tracerId =>
    state.tracers.filter(tracer => tracerId === tracer.id)[0]
}

const mutations = {
  setTracers: (state, { tracers }) => {
    state.tracers = tracers
  },
  setLoading: (state, loading) => {
    state.loading = loading
  },
  setSuccess: state => {
    state.success = true
    state.loading = false
    state.errors = []
  },
  setError: (state, { errors }) => {
    state.errors = errors
    state.loading = false
  },
  setDisplayFormat: (state, format) => {
    state.displayFormat = format
  },
  setPerPage: (state, perPage) => {
    state.perPage = perPage
  },
  setFilteredTracers: (state, filteredTracers) => {
    state.filteredTracers = filteredTracers
  }
}

const actions = {
  async loadTracers({ commit }) {
    try {
      commit('setLoading', true)

      const tracersRes = await api.get(`/tracers`)
      const tracers = tracersRes.data
      commit('setTracers', { tracers })
      commit('setFilteredTracers', tracers.map(t => t.id))
      commit('setSuccess')
    } catch (e) {
      let errors = [e.message]
      commit('setError', { errors })
    }
  }
}

export default {
  namespaced: true,
  state,
  getters,
  actions,
  mutations
}
