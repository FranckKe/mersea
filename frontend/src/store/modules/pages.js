import axios from 'axios'
import i18n from '../../i18n'
import slugify from 'slugify'

const api = axios.create({
  baseURL: process.env.VUE_APP_API_URL,
  headers: { Accept: 'application/json', 'Content-Type': 'application/json' }
})

const state = {
  pages: [],
  loading: true,
  success: false,
  errors: []
}
const getters = {
  getLoading: state => {
    return state.loading
  },
  getError: state => {
    return state.errors
  },
  getSuccess: state => {
    return state.success
  },
  getPageContent: state => {
    return pageInfo => {
      return (
        state.pages.filter(page => page.slug === pageInfo.pageName)[0] || {}
      ).content
    }
  },
  getCategories: state => {
    return [
      ...new Set(
        state.pages
          .map(page => page.category)
          .filter(page => page.category !== 'other')
      )
    ]
  },
  getAllPagesByCategory: state => {
    let categories = [
      ...new Set(
        state.pages
          .map(page => page.category)
          .filter(page => page.category !== 'other')
      )
    ]
    return categories.map(function(category) {
      return {
        category: slugify(category),
        pagesName: state.pages
          .filter(
            page => page.category === category && page.language === i18n.locale
          )
          .map(page => {
            return { raw: page.name, slug: page.slug }
          })
      }
    })
  },
  getPagesByCategory(state) {
    return category => {
      return state.pages
        .filter(
          page => page.category === category && page.language === i18n.locale
        )
        .map(page => {
          return { raw: page.name, slug: page.slug }
        })
    }
  }
}
const mutations = {
  setPages: (state, { pages }) => {
    state.pages = pages
  },
  setSuccess: state => {
    state.success = true
    state.loading = false
    state.errors = []
  },
  setError: (state, { errors }) => {
    state.errors = errors
    state.loading = false
  }
}
const actions = {
  async loadPages({ commit }) {
    try {
      const pagesData = await api.get(`/pages`)
      const pages = pagesData.data
      commit('setPages', { pages })
      commit('setSuccess')
    } catch (e) {
      let errors = [e.message]
      commit('setError', { errors })
    }
  }
}

export default {
  namespaced: true,
  state,
  getters,
  actions,
  mutations
}
