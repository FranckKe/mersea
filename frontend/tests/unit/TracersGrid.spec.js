import VueTestUtils from '@vue/test-utils'
import TracersGrid from '@/components/TracersGrid.vue'
import Vuex from 'vuex'
import pages from '../../src/store/modules/pages'
import tracers from '../../src/store/modules/tracers'
import Buefy from 'buefy'
import VueRouter from 'vue-router'
import moment from 'moment'
import tracersData from './data/tracers'
import i18n from '../../src/i18n'

VueTestUtils.config.mocks.$t = key => key
VueTestUtils.config.mocks.$tc = (key, count) => (key, count)

const localVue = VueTestUtils.createLocalVue()
localVue.use(Buefy)
localVue.use(Vuex)
localVue.use(VueRouter)

localVue.filter('formatDate', value =>
  moment(String(value)).format('MM/DD/YYYY')
)

const router = new VueRouter()

describe('Page.vue', () => {
  let store

  beforeEach(() => {
    store = new Vuex.Store({
      modules: {
        pages: pages,
        tracers: tracers
      }
    })
  })

  it('matches snapshot', () => {
    const wrapper = VueTestUtils.shallowMount(TracersGrid, {
      localVue,
      store,
      router,
      i18n,
      propsData: {
        tracers: tracersData
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
