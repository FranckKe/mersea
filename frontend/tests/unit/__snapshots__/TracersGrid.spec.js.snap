// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page.vue matches snapshot 1`] = `
<div class="columns is-multiline is-mobile">
  <div class="column is-half-mobile is-one-third-tablet is-one-quarter-desktop">
    <article class="card">
      <div class="card-image">
        <router-link-stub to="/tracers/d791becc-4033-4a92-b83d-6bc572a91266" tag="a" event="click">
          <div class="img-wrapper" style="background-image: url(undefined/rails/active_storage/representations/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaEpJaWsyTURObU5EZzFZUzFpTVRabUxUUm1OMlF0WW1VMk9DMWhOamxtT0dFNFpqQXdOV0VHT2daRlZBPT0iLCJleHAiOm51bGwsInB1ciI6ImJsb2JfaWQifX0=--5a84c3ccceb38261d47d65c68d9528dd9de1140c/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaDdCam9MY21WemFYcGxTU0lOTWpVMmVESTFOajRHT2daRlZBPT0iLCJleHAiOm51bGwsInB1ciI6InZhcmlhdGlvbiJ9fQ==--22c880aec9fc71affcdbe98c951b0b00c24a9ea6/avatar.png);"></div>
        </router-link-stub>
      </div>
      <div class="card-content">
        <div class="content">
          <p class="title is-4">
            <router-link-stub to="/tracers/d791becc-4033-4a92-b83d-6bc572a91266" tag="a" event="click">TestActiveStorage</router-link-stub>
          </p>
          <p>TestActiveStorage</p>
          <p><strong>Origin: </strong>TestActiveStorage</p>
          <p><strong>Type: </strong>TestActiveStorage</p>
          <p><strong>Created at: </strong>
            <time datetime="2016-1-1">06/29/2018</time>
          </p>
        </div>
      </div>
    </article>
  </div>
  <div class="column is-half-mobile is-one-third-tablet is-one-quarter-desktop">
    <article class="card">
      <div class="card-image">
        <router-link-stub to="/tracers/42941860-3a23-4187-a47a-2915c5505249" tag="a" event="click">
          <div class="img-wrapper" style="background-image: url(undefined/rails/active_storage/representations/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaEpJaWsxTW1KaE16QTBOeTAwWTJJM0xUUmtaR1l0WVRZd01pMDNZVEUyTVdFeFlUZGpNVE1HT2daRlZBPT0iLCJleHAiOm51bGwsInB1ciI6ImJsb2JfaWQifX0=--6f0e7cdcfe492745f0e65abefd2865ee33c16f2f/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaDdCam9MY21WemFYcGxTU0lOTWpVMmVESTFOajRHT2daRlZBPT0iLCJleHAiOm51bGwsInB1ciI6InZhcmlhdGlvbiJ9fQ==--22c880aec9fc71affcdbe98c951b0b00c24a9ea6/avatar.png);"></div>
        </router-link-stub>
      </div>
      <div class="card-content">
        <div class="content">
          <p class="title is-4">
            <router-link-stub to="/tracers/42941860-3a23-4187-a47a-2915c5505249" tag="a" event="click">Tracer 1</router-link-stub>
          </p>
          <p>aa</p>
          <p><strong>Origin: </strong>aa</p>
          <p><strong>Type: </strong>aa</p>
          <p><strong>Created at: </strong>
            <time datetime="2016-1-1">03/21/2018</time>
          </p>
        </div>
      </div>
    </article>
  </div>
  <div class="column is-half-mobile is-one-third-tablet is-one-quarter-desktop">
    <article class="card">
      <div class="card-image">
        <router-link-stub to="/tracers/7ff25eb9-a7bd-4c57-831f-48857fb6019e" tag="a" event="click">
          <div class="img-wrapper" style="background-image: url(undefined/rails/active_storage/representations/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaEpJaWxtWWpaa01qRXhPUzAxTURjMUxUUmhOVEF0WWpRM055MDFPV016Tmprek1HWXdabVlHT2daRlZBPT0iLCJleHAiOm51bGwsInB1ciI6ImJsb2JfaWQifX0=--22b1ec5462d0bcc0dc58dfa6440ed17a216cd250/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaDdCam9MY21WemFYcGxTU0lOTWpVMmVESTFOajRHT2daRlZBPT0iLCJleHAiOm51bGwsInB1ciI6InZhcmlhdGlvbiJ9fQ==--22c880aec9fc71affcdbe98c951b0b00c24a9ea6/blaze-2178749_1920.jpg);"></div>
        </router-link-stub>
      </div>
      <div class="card-content">
        <div class="content">
          <p class="title is-4">
            <router-link-stub to="/tracers/7ff25eb9-a7bd-4c57-831f-48857fb6019e" tag="a" event="click">Tracer 2</router-link-stub>
          </p>
          <p>test</p>
          <p><strong>Origin: </strong>test</p>
          <p><strong>Type: </strong>test</p>
          <p><strong>Created at: </strong>
            <time datetime="2016-1-1">03/21/2018</time>
          </p>
        </div>
      </div>
    </article>
  </div>
  <div class="column is-half-mobile is-one-third-tablet is-one-quarter-desktop">
    <article class="card">
      <div class="card-image">
        <router-link-stub to="/tracers/2c1a136d-a22e-4833-9cb8-2a880c77f2ee" tag="a" event="click">
          <div class="img-wrapper" style="background-image: url(undefined/rails/active_storage/representations/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaEpJaWswT0dVeU1qbGtNUzFsTldJd0xUUTFNV0V0T0RreU5DMDFNMk5sTkRVNFpqVmtaamtHT2daRlZBPT0iLCJleHAiOm51bGwsInB1ciI6ImJsb2JfaWQifX0=--d61ffa72a5b214319321038d34e17c88f6274ea5/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaDdCam9MY21WemFYcGxTU0lOTWpVMmVESTFOajRHT2daRlZBPT0iLCJleHAiOm51bGwsInB1ciI6InZhcmlhdGlvbiJ9fQ==--22c880aec9fc71affcdbe98c951b0b00c24a9ea6/salle_de_bain_sous_combles-ploudaniel.jpg);"></div>
        </router-link-stub>
      </div>
      <div class="card-content">
        <div class="content">
          <p class="title is-4">
            <router-link-stub to="/tracers/2c1a136d-a22e-4833-9cb8-2a880c77f2ee" tag="a" event="click">Tracer3</router-link-stub>
          </p>
          <p>Tracer3</p>
          <p><strong>Origin: </strong>Tracer3</p>
          <p><strong>Type: </strong>Tracer3</p>
          <p><strong>Created at: </strong>
            <time datetime="2016-1-1">05/15/2018</time>
          </p>
        </div>
      </div>
    </article>
  </div>
</div>
`;
